crates/matrix-sdk-crypto/src/session_manager/sessions.rs

```diff
@@ -514,6 +514,10 @@ impl SessionManager {
         let mut new_sessions: BTreeMap<&UserId, BTreeMap<&DeviceId, SessionInfo>> = BTreeMap::new();
         let mut store_transaction = self.store.transaction().await;
 
+        // Cache of our own device keys, so we only need to fetch it once, if
+        // we need it at all.
+        let mut our_device_keys: Option<DeviceKeys> = None;
```

Comment:
Could we actually fetch the device keys up front to avoid starting off with this empty and filling it in later? It might make this easier to read but I assume there's a good reason we're doing it lazily?

Reply:
If there are no sessions to save, then we don't need to fetch the device keys.  So it saves us from doing work if it isn't needed.  I don't know how much work it is to fetch the device keys, so I don't know if it makes a difference.

Reply:
With the new function for getting our device from storage, I decided to just fetch the keys every time -- we're already fetching the account every time.